File: src/main/java/com/daima/exthelp/Exp/SExp/StringExpression.java
package com.daima.exthelp.Exp.SExp;

public class StringExpression implements IStringExpression {
    private String str;

    // 构造方法
    public StringExpression(String str) {
        this.str = str;
    }

    // 方法1：输出字符串首字母
    public String getFirst() {
        if (str == null || str.isEmpty()) {
            return "";
        }
        return String.valueOf(str.charAt(0));
    }

    // 方法2：根据一个字符查询，并输出从字符开始到查询到目标字符的所有字符
    public String getSubstringFromEnd(char endChar) {
        if (str == null || str.isEmpty()) {
            return "";
        }


        int endIndex = str.indexOf(endChar);
        if (endIndex == -1) {
            return "";
        }
        return str.substring(0, endIndex + 1);
    }

    // 方法3：删除左侧若干长度的字符串
    public void removeLeftCharacters(int length) {
        if (str == null || str.isEmpty()) {
            str = "";
            return;
        }

        if (length < 0) {
            throw new IllegalArgumentException("Length cannot be negative");
        }

        if (length >= str.length()) {
            str = "";
        } else {
            str = str.substring(length);
        }
    }

    // 获取当前字符串（为了方便展示结果，添加一个getter方法）
    public String getString() {
        return str;
    }
}

File: src/main/java/com/daima/exthelp/Exp/PExp/StringExpression.kt
package com.daima.exthelp.Exp.PExp

class StringExpression(private var str: String?) : IStringExpression {

    // 方法1：输出字符串首字母
    override fun getFirst(): String {
        return if (str.isNullOrEmpty()) {
            ""
        } else {
            str!![0].toString()
        }
    }

    // 方法2：根据一个字符查询，并输出从字符开始到查询到目标字符的所有字符
    override fun getSubstringFromEnd(endChar: Char): String {
        if (str.isNullOrEmpty()) {
            return ""
        }
        val endIndex = str!!.indexOf(endChar)
        return if (endIndex == -1) {
            ""
        } else {
            str!!.substring(0, endIndex + 1)
        }
    }

    // 方法3：删除左侧若干长度的字符串
    override fun removeLeftCharacters(length: Int) {
        if (str.isNullOrEmpty()) {
            str = ""
            return
        }

        require(length >= 0) { "Length cannot be negative" }

        str = if (length >= str!!.length) {
            ""
        } else {
            str!!.substring(length)
        }
    }
    override fun getString(): String {
        return str!!
    }
}